symfony/plugins/orangehrmCorePlugin/Api/V2/CollectionEndpointInterface.php

```diff
@@ -0,0 +1,45 @@
+<?php
+/**
+ * OrangeHRM is a comprehensive Human Resource Management (HRM) System that captures
+ * all the essential functionalities required for any enterprise.
+ * Copyright (C) 2006 OrangeHRM Inc., http://www.orangehrm.com
+ *
+ * OrangeHRM is free software; you can redistribute it and/or modify it under the terms of
+ * the GNU General Public License as published by the Free Software Foundation; either
+ * version 2 of the License, or (at your option) any later version.
+ *
+ * OrangeHRM is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY;
+ * without even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
+ * See the GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License along with this program;
+ * if not, write to the Free Software Foundation, Inc., 51 Franklin Street, Fifth Floor,
+ * Boston, MA  02110-1301, USA
+ */
+
+namespace OrangeHRM\Core\Api\V2;
+
+use OrangeHRM\Core\Api\V2\Serializer\EndpointCreateResult;
+use OrangeHRM\Core\Api\V2\Serializer\EndpointDeleteResult;
+use OrangeHRM\Core\Api\V2\Serializer\EndpointGetAllResult;
+
+interface CollectionEndpointInterface
+{
+    /**
+     * Get collection of resources
+     * @return EndpointGetAllResult
+     */
+    public function getAll(): EndpointGetAllResult;
+
+    /**
+     * Create a new resource
+     * @return EndpointCreateResult
+     */
+    public function create(): EndpointCreateResult;
```

Comment:
Create / delete not part of CollectionEndpoint any way .. use CrudEndPoint kind of name

Reply:
Why create added to collection; since before create it don't have unique id to identify. So have to call POST `/api/v2/admin/job-titles`, not like POST `/api/v2/admin/job-titles/{id}`

Why delete added to collection; delete can handle in both ways, collection as well as resource. Best way is handle as resource. But in case we need bulk deletions, we need delete array of resources. then we need delete on collection. Haiving delete in both resource and collection is costly, so move to collection

Reply:
As I understand 
GET: `/api/v2/admin/job-title/{id}`  getting a single job title 
POST: `/api/v2/admin/job-title`  save a single/multi job titles
POST: `/api/v2/admin/job-title/{id}` update job title
DELETE: `/api/v2/admin/job-title/{id}` delete job title
GET: `/api/v2/admin/job-title/?query` Search job titles